src/libertem/udf/base.py

```diff
@@ -1148,26 +1373,47 @@ def get_tiling_preferences(self) -> TilingPreferences:
         }
 
     def get_backends(self) -> BackendSpec:
-        # TODO see interaction with C++ CUDA modules requires a different type than CuPy
         '''
         Signal which computation back-ends the UDF can use.
 
-        :code:`numpy` is the default CPU-based computation.
-
-        :code:`cuda` is CUDA-based computation without CuPy.
-
-        :code:`cupy` is CUDA-based computation through CuPy.
+        Allowed are values in :attr:`libertem.udf.base.UDF.BACKEND_ALL`.
+        :code:`(UDF.BACKEND_NUMPY, )` is returned by default for CPU-based computation.
 
         .. versionadded:: 0.6.0
 
+        .. versionchanged:: 0.11.0
+            Extended from just NumPy vs. CuPy to include more generic specification of backends,
+            including sparse arrays. See :ref:`sparse` for details! The backend specifications are
+            backwards-compatible with the previously supported values.
+
         Returns
         -------
 
-        backend : Iterable[str]
-            An iterable containing possible values :code:`numpy` (default), :code:`'cuda'` and
-            :code:`cupy`
+        backends
+            A single value or iterable containing values from the supported backends
+            in :attr:`libertem.udf.base.UDF.BACKEND_ALL`
+        '''
+        return (UDF.BACKEND_NUMPY, )
+
+    def forbuf(self, arr, target):
         '''
-        return ('numpy', )
+        Convert array to format that is compatible with result buffers and
+        reshape.
+
+        This function should be wrapped around assignment to result buffers if
+        the array :code:`arr` might be sparse and/or have a flattened sig
+        dimension. It converts any of the supported sparse input formats into
+        the array format of the target buffer. If the argument is already matching the buffer
+        array or is not recognized, it is a no-op.
+
+        The result of array operations on a sparse input tile can't always be
+        merged into dense result buffers directly since arrays from the
+        :mod:`sparse` package are not converted to NumPy arrays automatically.
+
```

Comment:
As this is one of the more important user-facing functions added, it would be great to have a "Parameters" section here that describes the types, and maybe a small example code snippet.

Reply:
Good point!